File: java/01423_maximum_points_you_can_obtain_from_cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        int n = cardPoints.length;

        int currentScore = 0;
        for (int i = 0; i < k; i++) {
            currentScore += cardPoints[i];
        }

        int maxScore = currentScore;

        for (int i = 0; i < k; i++) {
            currentScore += cardPoints[n - 1 - i];
            currentScore -= cardPoints[k - 1 - i];
            maxScore = Math.max(maxScore, currentScore);
        }

        return maxScore;
    }
}

// Time complexity - O(k)
// Space complexity - O (1)

File: python/01423_maximum_points_you_can_obtain_from_cards.py

class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)

        currentScore = sum(cardPoints[:k])
        maxScore = currentScore

        for i in range(k):
            currentScore += cardPoints[n - 1 - i]
            currentScore -= cardPoints[k - 1 - i]
            maxScore = max(maxScore, currentScore)

        return maxScore

# Time complexity - O(k)
# Space complexity - O (1)